blocks/store/actions.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { castArray } from 'lodash';
+
+/**
+ * Returns an action object used in signalling that block types have been added.
+ *
+ * @param {Array|Object} blockTypes Block types received.
+ *
+ * @return {Object} Action object.
+ */
+export function addBlockTypes( blockTypes ) {
+	return {
+		type: 'ADD_BLOCK_TYPES',
+		blockTypes: castArray( blockTypes ),
```

Comment:
I like the fact that this action receives multiple blocks. It opens the door to bulk registration, instead of dispatching one action per register block.

But, right now we call registerBlockType per block so an action is dispatched per block and at the start, we are dispatching many actions. Right after landing this, I think we should change our registration mechanism to make use of the possibility of using ADD_BLOCK_TYPES with multiple blocks. If a general approach does not look feasible at least, we can change our registerCoreBlocks to register all core blocks with a single action.